src/datanode/src/tests/instance_test.rs

```diff
@@ -342,6 +342,104 @@ pub async fn test_execute_create() {
     assert!(matches!(output, Output::AffectedRows(0)));
 }
 
+#[tokio::test]
+async fn test_rename_table() {
+    let instance = MockInstance::new("test_rename_table_local").await;
+
+    let output = execute_sql(&instance, "create database db").await;
+    assert!(matches!(output, Output::AffectedRows(1)));
+
+    let output = execute_sql_in_db(
+        &instance,
+        "create table demo(host string, cpu double, memory double, ts timestamp, time index(ts))",
+        "db",
+    )
+    .await;
+    assert!(matches!(output, Output::AffectedRows(0)));
+
+    let output = execute_sql_in_db(&instance, "show tables", "db").await;
```

Comment:
This test has some redundant codes, I think we could wrap `show tables` and `select` into a helper function. It's ok to test insert first.
```rust
// make sure table insertion is ok after altered table name
let output = execute_sql_in_db(
    &instance,
    "insert into test_table(host, cpu, memory, ts) values ('host2', 2.2, 200, 2000)",
    "db",
)
.await;
assert!(matches!(output, Output::AffectedRows(1)));

check_show_tables_and_select().await;

instance.restart().await.unwrap();

check_show_tables_and_select().await;
```